resources/assets/lib/chat/conversation-view.tsx

```diff
@@ -192,53 +193,59 @@ export default class ConversationView extends React.Component<Props> {
       return <div className='chat-conversation' />;
     }
 
+    const renderInput = !(!channel.canMessage && channel.type === 'ANNOUNCE');
```

Comment:
this is a rather difficult to read conditional... (was it originally `hideInput` or something)

```suggestion
    const renderInput = channel.canMessage || channel.type !== 'ANNOUNCE';
```